File: basic_program.f90
program main

  implicit none
  integer i, n
  real*8 sum, exact
  real*8, allocatable :: x(:)

  n=100
  allocate(x(n))

  !$omp parallel do
  do i=1,n
     x(i) = i
  end do

  sum = 0
  !$omp parallel do reduction(+: sum)
  do i=1,n
     sum = sum + x(i)
  end do

  exact = n*(n+1)/2.0d0;

  if( abs(1.0d0 - sum/exact) > 1.0d-15 ) then
     write(*, '(A,ES22.15,A,ES22.15)') "Error: sum does not match", sum, " !=", exact
     call exit(1)
  end if

  deallocate(x)

  write(*, '(A)') "Finished test with no errors"
  
end program main

File: basic_program.cc
#include <iostream>

int main() {

  std::cout << "Beginning C++ basic program...\n";

  int n = 1000;
  double *x = new double[n];

#pragma omp parallel for
  for(int i=0;i<n;i++)
    x[i] = i+1;

  double sum = 0;
#pragma omp parallel for reduction(+: sum)
  for(int i=0;i<n;i++)
    sum += x[i];

  double exact = n*(n+1)/2.0;

  if( std::abs(1.0 - sum/exact) > 1e-15 ) {
    std::cerr.precision(15);
    std::cerr << std::scientific << "Sum does not match: " << sum << " != " << exact << std::endl;
    exit(1);
  }

  delete[] x;

  std::cout << "Finished test with no errors\n";

  return 0;
}
